src/imports/register-extensions.js

```diff
@@ -53,9 +71,15 @@ const registerExtensions = (keyword, object, includeGlobal = false,
       const extensionPath = path.dirname(packageJsonPath);
       const extension = require(extensionPath); // eslint-disable-line global-require
 
-      extension[registerFunction](object, dependencyPath);
+      if (extension[registerFunction]) {
+        registeredExtension = extension[registerFunction](object, keyword);
+      } else {
+        logger.warn(`No register function found for extension ${dependency}`);
+      }
```

Comment:
Also if we are allowing for this function to have any name then we are assuming it may be used to find other kinds of functions besides register so should the error message be something more like `\`No function with the name ${registerFunction} is found for extension ${dependency}\``